authentication_service/middleware/middleware.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	helper "github.com/leechongyan/Studtor_backend/authentication_service/helpers/account"
+
+	"github.com/gin-gonic/gin"
+)
+
+func Authentication() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		clientToken := c.Request.Header.Get("Authorization")
+		if clientToken == "" {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": fmt.Sprintf("No Authorization Header provided")})
```

Comment:
should we return 401 unauthorized instead? also, idt the `fmt.Sprintf` does anything